haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
```

Comment:
@kevinconway, please document the exceptions that this method may reasonably raise, so callers know what to expect. This will probably be something like this:

```
    def connect(self, (host, port), klass=socket.socket):
        '''
        Connect assuming a host and port tuple.

        :raises socket.gaierror: error resolving address
        :raises socket.error: error configuring or connecting socket
        '''
```